code/script.js

```diff
@@ -1,47 +1,247 @@
+
+    inputWrapper.innerHTML = `
+        <button id="neroDavola">Nero d'Avola</button>
+        <button id="barbera">Barbera</button>
+        <button id="amarone">Amarone</button>
+    `
+
+    document.getElementById('neroDavola').addEventListener('click', () => {
+        showMessage("Nero d'Avola", 'user');
+        setTimeout(() => wineRecoNero(), 2000);
+    })
+
+    document.getElementById('barbera').addEventListener('click', () => {
+        showMessage('Barbera', 'user');
+        setTimeout(() => wineRecoBarbera(), 2000);
+    })
+
+    document.getElementById('amarone').addEventListener('click', () => {
+        showMessage('Amarone', 'user');
+        setTimeout(() => wineRecoAmarone(), 2000);
+    })
+};
+
+
+// 4.1.1    Italy - bot gives the last message based on user's choice on grape
+const wineRecoNero = () => {
+    showMessage(`Fantastic choice! The finest Nero d'Avola can be found on Sicily! Enjoy!`, 'bot');
+    sound.play();
+    inputWrapper.innerHTML = ``;
+}
+
+const wineRecoBarbera = () => {
+    showMessage(`An amazing grape! It's the most planted grape in all of Italy! Enjoy!`, 'bot');
+    sound.play();
+    inputWrapper.innerHTML = ``;
+}
+
+const wineRecoAmarone = () => {
+    showMessage(`Exclusive taste, eh? It's worth every penny! Enjoy!`, 'bot');
+    sound.play();
+    inputWrapper.innerHTML = ``;
+}
+
+
+// 4.2    Spain - user chooses the preferred grape from Spain
+const questionGrapeSpain = () => {
+    showMessage(`Perfect! Might as well! Which of the following grapes to you prefer?`, 'bot');
+
+    inputWrapper.innerHTML = `
+        <button id="tempranillo">Tempranillo</button>
+        <button id="granache">Granache</button>
+        <button id="syrah">Syrah</button>
+    `
+
+    document.getElementById('tempranillo').addEventListener('click', () => {
+        showMessage('Tempranillo', 'user');
+        setTimeout(() => wineRecoTemp(), 2000);
+    })
+
+    document.getElementById('granache').addEventListener('click', () => {
+        showMessage('Granache', 'user');
+        setTimeout(() => wineRecoGran(), 2000);
+    })
+
+    document.getElementById('syrah').addEventListener('click', () => {
+        showMessage('Syrah', 'user');
+        setTimeout(() => wineRecoSyrah(), 2000);
+    })
+};
+
+
+// 4.2.1    Spain - bot gives the last message based on user's choice on grape
+const wineRecoTemp = () => {
+    showMessage(`Woho! It´s the best known quality red wine grape in Spain. Enjoy!`, 'bot');
+    sound.play();
+    inputWrapper.innerHTML = ``;
+}
+
+const wineRecoGran = () => {
+    showMessage(`Yummie! A very fruity wine with lots of raspberry aromas. Enjoy!`, 'bot');
+    sound.play();
+    inputWrapper.innerHTML = ``;
+}
+
+const wineRecoSyrah = () => {
+    showMessage(`You´re in for a treat! A full bodied wine from Galicia. Enjoy!`, 'bot');
+    sound.play();
+    inputWrapper.innerHTML = ``;
+}
```

Comment:
It can also be executed by the same tweaked function I was writing about earlier 🤓